src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -105,11 +103,67 @@ private async Task PromoteBuildAsync(int PromoteToMaestroChannelId)
                 $"\"SDLValidationContinueOnError\": \"{ _options.SDLValidationContinueOnError }\", " +
                 $"}}";
 
-            await azdoClient.StartNewBuildAsync(BuildPromotionPipelineAccountName, 
+            var azdoBuildId = await azdoClient.StartNewBuildAsync(BuildPromotionPipelineAccountName, 
                 BuildPromotionPipelineProjectName, 
                 BuildPromotionPipelineId, 
+                arcadeSDKSourceBranch,
+                arcadeSDKSourceSHA,
                 queueTimeVariables)
                 .ConfigureAwait(false);
+
+            Console.WriteLine($"Build {build.Id} will be assigned to channel '{targetChannel.Name}' once this promotion build finishes: " +
+                $"https://{BuildPromotionPipelineAccountName}.visualstudio.com/{BuildPromotionPipelineProjectName}/_build/results?buildId={azdoBuildId}&view=results");
+        }
+
+        /// <summary>
+        /// By default the source branch/SHA for the Build Promotion pipeline will be the branch/SHA
+        /// that produced the Arcade.SDK used by the build being promoted. The user can override that
+        /// by specifying both, channel & SHA, on the command line.
+        /// </summary>
+        /// <param name="build">Build for which the Arcade SDK dependency build will be inferred.</param>
+        private async Task<(string sourceBranch, string sourceVersion)> GetSourceBranchInfoAsync(Build build)
+        {
+            if (!string.IsNullOrEmpty(_options.SourceBranch) && !string.IsNullOrEmpty(_options.SourceSHA))
+            {
+                return (_options.SourceBranch, _options.SourceSHA);
+            }
+
+            IGitRepo targetBuildRepoClient = string.IsNullOrEmpty(build.GitHubRepository) ?
+                (IGitRepo) new AzureDevOpsClient(_options.GitLocation, _options.AzureDevOpsPat, Logger, temporaryRepositoryPath: null) :
+                (IGitRepo) new GitHubClient(_options.GitLocation, _options.GitHubPat, Logger, temporaryRepositoryPath: null, cache: null);
+
+            IRemote barRemote = RemoteFactory.GetBarOnlyRemote(_options, Logger);
+
+            var gitFileManager = new GitFileManager(targetBuildRepoClient, Logger);
+
+            var sourceBuildDependencies = await gitFileManager.ParseVersionDetailsXmlAsync(
+                string.IsNullOrEmpty(build.GitHubRepository) ? 
+                    build.AzureDevOpsRepository : 
+                    build.GitHubRepository, 
+                build.Commit);
+
+            var sourceBuildArcadeSDKDependency = sourceBuildDependencies.FirstOrDefault(i => string.Equals(i.Name, "Microsoft.DotNet.Arcade.Sdk", StringComparison.OrdinalIgnoreCase));
+
+            if (sourceBuildArcadeSDKDependency == null)
+            {
+                throw new ArgumentException("You're trying to promote a build that doesn't have a dependency on Microsoft.DotNet.Arcade.Sdk.");
```

Comment:
I would prefer these be clean console writes, then returns of the error code. As it is right now, it will print a generic error with this message inside it, which is not the pattern we have been following here for command line issues in Darc.